### ISSUE TITLE
Mimalloc fails to run

### ISSUE THREAD

**Issue by olivierpierre** [ssrg-vt/SlimGuard#2]
[Mimalloc](https://github.com/microsoft/mimalloc)'s test-stress is a multi-threaded microbenchmark for memory allocators. Its default configuration fails to run with SlimGuard. To reproduce the issue, see the `make run` target [here](https://github.com/ssrg-vt/SlimGuard/tree/master/test/mimalloc):

```shell
make run
gcc -g test-stress.c -o test-stress -lpthread
LD_PRELOAD=/home/pierre/Desktop/SlimGuard/test/mimalloc/../../libSlimGuard.so ./test-stress
Using 32 threads with a 10% load-per-thread and 50 iterations
- iterations left:  40
- iterations left:  30
- iterations left:  20
Segmentation fault
make: *** [Makefile:13: run] Error 139
```

**Comment by olivierpierre** [comment]
It seems to happen only when the number of thread (see the `THREADS` global variable in `test-stress.c`) is greater than the number of cores on the host machine.

**Comment by olivierpierre** [comment]
Fixed in 70ada6f3e5b6f6492a49cdee7b684c3994804100.
